robottelo/ui/computeresource.py

```diff
@@ -1,72 +1,116 @@
-                    region = self.find_element(locators["resource.region"])
-                    Select(region).select_by_visible_text(region)
-                elif provider_type == "Openstack":
-                    tenant = self.find_element(
-                        locators["resource.rhos_tenant"])
-                    Select(tenant).select_by_visible_text(tenant)
-            if provider_type == "Libvirt":
-                if self.wait_until_element(locators["resource.url"]):
-                    self.find_element(locators["resource.url"]).send_keys(url)
-        if libvirt_display is not None:
-            display = self.find_element(locators["resource.libvirt_display"])
-            Select(display).select_by_visible_text(libvirt_display)
-        if libvirt_set_passwd is False:
-            self.find_element(
-                locators["resource.libvirt_console_passwd"]).click()
-        self.find_element(locators["resource.test_connection"]).click()
-        self.wait_for_ajax()
+            Select(
+                self.find_element(locators["resource.provider_type"])
+            ).select_by_visible_text(provider_type)
+        for parameter_name, parameter_value, parameter_type in parameter_list:
+            param_locator = '.'.join((
+                'resource',
+                (parameter_name.lower()).replace(' ', '_')
+            ))
+            self.wait_until_element(locators[param_locator])
+            if parameter_type == 'field':
+                self.find_element(
+                    locators[param_locator]).send_keys(parameter_value)
+            elif parameter_type == 'select':
+                Select(
+                    self.find_element(locators[param_locator])
+                ).select_by_visible_text(parameter_value)
+            elif parameter_type == 'checkbox':
+                if (self.find_element(locators[param_locator]).is_selected() !=
+                        parameter_value):
+                    self.click(locators[param_locator])
+            elif parameter_type == 'special select':
+                button_locator = '.'.join((
+                    'resource',
+                    (parameter_name.lower()).replace(' ', '_'),
+                    'button'
+                ))
+                self.click(locators[button_locator])
+                self.find_element(
+                    locators[param_locator]
+                ).select_by_visible_text(parameter_value)
+
+    def _configure_orgs(self, orgs, org_select):
+        """Provides configuration capabilities for compute resource
+        organization. The following format should be used::
 
-    def create(self, name, orgs, description=None, org_select=True,
-               provider_type=None, url=None, user=None,
-               password=None, region=None, libvirt_display=None,
-               libvirt_set_passwd=True, tenant=None):
+            orgs=['Aoes6V', 'JIFNPC'], org_select=True
+
+        """
+        if orgs:
```

Comment:
Optional arguments in python have a default value [1]. I think we can add a default value of None and if it is not None do the job. Also make that explicitly in docstring.

[1] http://www.diveintopython.net/power_of_introspection/optional_arguments.html